syntax = "proto3";

package sd3;


message Response{
    enum Tipo{
        DISPOSITIVOS = 0;
        RESPOSTA = 1;
    }

    Tipo tipo = 1;
    repeated Dispositivo dispositivos = 2;
    string id_dispositivos = 3;
    string resposta = 4;
}

message Request{

    enum Tipo{
        DESCOBERTA = 0;
        OPERACAO = 1;
    }

    Tipo tipo = 1;
    string id_dispositivo = 2;
    string operacao = 3;
}

message GatewayDispositivo{

    enum Tipo{
        DESCOBERTA = 0;
        DISPOSITIVO = 1;
        OPERACAO = 2;
        RESPOSTA = 3;
        ANUNCIO =4;
    }

    Tipo tipo = 1;
    Dispositivo dispositivo = 2;
    string id_dispositivo = 3;
    string operacao = 4;
    string resposta = 5;

}

message Dispositivo{

    string nome = 1;
    string id = 2;
    repeated string operacoes = 3;
    
}

